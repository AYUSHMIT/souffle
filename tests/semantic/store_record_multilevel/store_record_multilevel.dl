// Souffle - A Datalog Compiler
// Copyright (c) 2020, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// Test writing of nested records

.type Seq = [ first : number, second : Seq ]

.decl A(s: Seq)
A(nil).
A([5, [4, [3, nil]]]).
A([-8, [-1, [2, [3, nil]]]]).


.output A(IO=stdout,headers=true)
