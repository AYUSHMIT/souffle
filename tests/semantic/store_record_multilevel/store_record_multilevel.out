---------------
A
s
===============
nil
[5, [4, [3, nil]]]
[-8, [-1, [2, [3, nil]]]]
===============
